### PR TITLE
Convert PHPDoc to return type hints

### PR DESCRIPTION
**Pull request by AJenbo**
Type: refactoring / documentation update
Breaking change: yes

Simply ran:
```php
vendor/bin/php-cs-fixer fix --rules=phpdoc_to_return_type
vendor/bin/php-cs-fixer fix
```

### CODE REVIEW

**Review by kylekatarnls — DISMISSED** [review]

**Comment by kylekatarnls** [comment]
I wonder how we can proceed such refactors dealing with conflicts. I guess we should do them by chunk (limited number of folders or files).

**Comment by AJenbo** [comment]
> I wonder how we can proceed such refactors dealing with conflicts. I guess we should do them by chunk (limited number of folders or files).

They are the last things missing so If we can just merge them as the next thing it should not be an issue :)
If we split them up they will still conflict, so unless that makes it easier for you to re-approve I don't think it's going to make it easier to land things.

File: src/main/php/PDepend/Source/AST/ASTNamespace.php [hunk 1]
```diff
@@ -199,9 +197,8 @@ public function getTypes()
      * Adds the given type to this namespace and returns the input type instance.
      *
      * @param AbstractASTClassOrInterface $type
-     * @return AbstractASTClassOrInterface
      */
-    public function addType(AbstractASTType $type)
+    public function addType(AbstractASTType $type): AbstractASTClassOrInterface
```
Comment (tvbeek) [inline src/main/php/PDepend/Source/AST/ASTNamespace.php:201]: I think the return type should be AbstractASTType

```suggestion
    public function addType(AbstractASTType $type): AbstractASTType 
```

Comment (AJenbo) [reply src/main/php/PDepend/Source/AST/ASTNamespace.php:201]: It's actually the input that should be `AbstractASTClassOrInterface`. You can see this by it adding the type to $this->types which is a collection of AbstractASTClassOrInterface.

The type was to loose in 2.x and we overwrote it with the PHPDoc, this should be corrected in the native type, but I would like to wait till we are done with the automated cleanups first.

**Review by tvbeek — APPROVED** [review]

File: src/main/php/PDepend/Source/AST/ASTNamespace.php (hunk 1 above)
Comment (AJenbo) [reply src/main/php/PDepend/Source/AST/ASTNamespace.php:201]: I added it to https://github.com/pdepend/pdepend/pull/866

**Comment by AJenbo** [comment]
@tvbeek thanks for reviewing, I added the thing you pointed out to https://github.com/pdepend/pdepend/pull/866

**Pull request merged**